pkg/kubelet/lifecycle/handlers.go

```diff
@@ -99,6 +99,9 @@ func resolvePort(portReference intstr.IntOrString, container *v1.Container) (int
 	}
 	for _, portSpec := range container.Ports {
 		if portSpec.Name == portName {
+			if portSpec.Protocol != v1.ProtocolTCP {
```

Comment:
The only caller of this happens to be runHTTPHandler today, but there's nothing in the function name that indicates this will only return TCP named ports.

Reply:
the default is TCP if unspecified, right? do static pod manifests loaded by the kubelet have defaults applied, or does this need to handle the possibility of an empty Protocol field meaning "TCP"?

Reply:
> the default is TCP if unspecified, right? do static pod manifests loaded by the kubelet have defaults applied, or does this need to handle the possibility of an empty Protocol field meaning "TCP"?

https://github.com/kubernetes/kubernetes/blob/4f9317596ccb9fcb37bf183babf3e8640afcca9b/staging/src/k8s.io/api/core/v1/types.go#L1837-L1841
The default value is set when deserializing here.

Reply:
> The only caller of this happens to be runHTTPHandler today, but there's nothing in the function name that indicates this will only return TCP named ports.

Good point!
Updated, i added protocol parameters to the function.

Reply:
> The only caller of this happens to be runHTTPHandler today, but there's nothing in the function name that indicates this will only return TCP named ports.

Good point!
Updated, i added protocol parameters to the function.